### ISSUE TITLE
Standalone kdl_parser library

### ISSUE THREAD

**Issue by christian-rauch** [ros/kdl_parser#12]
The `kdl_parser` is used in a couple of robotics applications outside of ROS. It would be nice, if the ROS and catkin dependencies would be removed or become optional (e.g. only use `catkin` and `rosconsole` if they are available).

The dependency on ROS is very low, since only the logging capabilities from `rosconsole` (`ROS_ERROR`, ...) are used. This could be disabled at compile time and just be stream to `cerr` if `rosconsole` is not available. Since the `urdf` package is now part of the Ubuntu repo, there is no need to use `catkin` to discover its location. A plain `CMakeLists.txt` could achieve the same.

Are there other outstanding issues, which prevent to disentangle `kdl_parser` from ROS?

**Comment by sloretz** [comment]
There isn't much preventing ROS from being an optional dependency. Would those applications not using ROS be fine with building `kdl_parser` from source?

The package could use tests that are not dependent on `rostest`. They should be built if either `BUILD_TESTING` or `CATKIN_ENABLE_TESTING` is set. If `catkin` is not available then the library should install a `kdl_parser-config.cmake` and `kdl_parser-config-version.cmake`.

If you would be willing to make the changes then it would make sense to start it from the `melodic-devel` branch. Might as well bump up the minimum cmake version since melodic requires [cmake 3.7.2 at the lowest](http://www.ros.org/reps/rep-0003.html#melodic-morenia-may-2018-may-2023).

**Comment by christian-rauch** [comment]
I created PR https://github.com/ros/kdl_parser/pull/13 for this feature. I did not bumped the cmake version to allow the standalone building on older platforms. Also, the tests are disabled on standalone builds since the CI is done on the ROS buildfarm anyway.
There are some minor drawbacks, since `kdl_parser` is depending on `urdf`, which itself is again depending on ROS for similar reasons (see PR).
